projects/igniteui-angular/src/lib/drop-down/drop-down.component.ts

```diff
@@ -215,7 +218,14 @@ export class IgxDropDownBase implements OnInit, IToggleView {
      */
     public get selectedItem(): any {
         const selection = this.selectionAPI.get_selection(this.id);
-        return selection && selection.length > 0 ? selection[0] as IgxDropDownItemComponent : null;
+        const selectedItem = selection && selection.length > 0 ? selection[0] as IgxDropDownItemComponent : null;
+        if (selectedItem) {
+            if (selectedItem.isSelected) {
+                return selectedItem;
+            }
+            this.selectionAPI.set_selection(this.id, []);
```

Comment:
Umm shouldn't this be somewhere in `changeSelectedItem` and/or the item check instead of this getter?
Then you only have `selectedItem && selectedItem.isSelected => return selectedItem` otherwise `null`.

Reply:
If you set isSelected = true to an item in the markup this will set the selectedItem in selection service. At this point if the developer set to the same item isSelected = false in code behind this will not force change detector and selected item in selection service will not update. At this point if the developer tries to get the selected item from the drop down he will end with item which isSelected is set to false. I do not see any other place where I can handle this absolutely edge scenario.